### DISCUSSION TITLE
Ajouter un pdf dans synapps

### DISCUSSION THREAD

**Discussion by Mathis554** [witsa/synapps#804] [General]
Bonjour,
J'aimerais ajouter un PDF à mon projet Synapps mais je ne sais pas comment m'y prendre
Merci par avance

**Comment by Raphwit** [comment]
Bonjour, où si situe votre PDF, en ligne ou en local ?

**Reply by Mathis554** [reply]
Bonjour,
Le pdf et sur mon ordinateur, il faut le mettre dans l'UC ?

**Reply by Raphwit** [reply]
Il faut effectivement le mettre dans l'UC et ensuite renseigner l'adresse où celle-ci se situe dans l'UC au sein d'une iFrame

**Reply by WITNicolas** [reply]
Bonjour, lien vers la doc 
https://witsa.github.io/synapps/tutorials/open-pdf/iframe-pdf.html

**Reply by Mathis554** [reply]
Bonjour, Merci beaucoup j'ai réussie

**Comment by Eregul56** [comment]
Bonjour,
Je me permets de rebondir sur cette question : peut-on afficher des PDF stockés sur un drive (Dropbox, Google Drive, etc.) ?
J'ai fait un test avec un acteur iframe mais comme je m'y attendais le fichier était bloqué :)
Merci.

**Reply by manufitoussiwit** [reply]
Bonjour,
Oui c'est possible en tout cas avec Google Drive.
Il suffit de : 
- générer un lien de partage sur le document dans google drive, 

![image](https://github.com/user-attachments/assets/84c07a0d-1abb-406a-8fa1-f4c2f3159ab9)

- l'inscrire dans l'adresse d'un acteur iframe
- activer toutes les options de l'acteur.
- modifier le lien : remplacer la fin de l'URL "view" par "preview"
par exemple :
`https://drive.google.com/file/d/1v-MKHSXX_CBAMwaC-hgBt5lIDYKcgmT6/view?usp=drive_link`
devient : 
`https://drive.google.com/file/d/1v-MKHSXX_CBAMwaC-hgBt5lIDYKcgmT6/preview`

un exemple de scène : 
```
SYNAPPS-STUDIO-SCENE|{"config":{"key":"scene131","name":"Exemple Doc Drive"},"leadActor":{"type":"layout/stack","key":"stack1","children":[{"type":"display/iframe","key":"iframe1","properties":{"height":"780px","width":"840px","url":"https://drive.google.com/file/d/1v-MKHSXX_CBAMwaC-hgBt5lIDYKcgmT6/preview","allowTopNavigation":true}}]}}
```
